PHPCI/Model/Build/SubversionBuild.php

```diff
@@ -0,0 +1,187 @@
+ *
+ * @copyright    Copyright 2014, Block 8 Limited.
+ * @license      https://github.com/Block8/PHPCI/blob/master/LICENSE.md
+ * @link         https://www.phptesting.org/
+ */
+
+namespace PHPCI\Model\Build;
+
+use PHPCI\Model\Build;
+use PHPCI\Builder;
+
+/**
+* Remote Subversion Build Model
+* @author       Nadir Dzhilkibaev <imam.sharif@gmail.com>
+* @package      PHPCI
+* @subpackage   Core
+*/
+class SubversionBuild extends Build
+{
+    /**
+    * Get the URL to be used to clone this remote repository.
+    */
+    protected function getCloneUrl()
+    {
+        return $this->getProject()->getReference();
+    }
+
+    /**
+    * Create a working copy by cloning, copying, or similar.
+    */
+    public function createWorkingCopy(Builder $builder, $buildPath)
+    {
+        $this->handleConfig($builder, $buildPath);
+        
+        $key = trim($this->getProject()->getSshPrivateKey());
+	
+        if (!empty($key)) {
+            $success = $this->cloneBySsh($builder, $buildPath);
+        } else {           
+            $success = $this->cloneByHttp($builder, $buildPath);
+        }
+
+        if (!$success) {
+            $builder->logFailure('Failed to export remote subversion repository.');
+            return false;
+        }
+
+        return $this->handleConfig($builder, $buildPath);
+    }
+
+    /**
+    * Use an HTTP-based svn export.
+    */
+    protected function cloneByHttp(Builder $builder, $cloneTo)
+    {
+        $cmd = 'svn export ';    
+        
+        $svn = $builder->getConfig('svn');
+        if (!is_null($svn)) {
+	    foreach ($svn as $key => $value)
+		$cmd .= ' --' . $key . ' ' . $value . ' ';
+        }
+        
+        $depth = $builder->getConfig('clone_depth');
+        
+        if (!is_null($depth)) {
+            $cmd .= ' --depth ' . intval($depth) . ' ';
+        }
+        
+        $cmd .= ' --non-interactive -q -r %s %s "%s"';
+
+        $success = $builder->executeCommand($cmd, $this->getBranch(), $this->getCloneUrl(), $cloneTo);
+
+        if ($success) {
+            $success = $this->postCloneSetup($builder, $cloneTo);
+        }
+
+        return $success;
+    }
+
+    /**
+    * Use an SSH-based svn export.
+    */
+    protected function cloneBySsh(Builder $builder, $cloneTo)
+    {
+        $keyFile = $this->writeSshKey($cloneTo);
+
+        if (!IS_WIN) {
+            $sshWrapper = $this->writeSshWrapper($cloneTo, $keyFile);
+        }
+
+        // Do the svn export:
+        $cmd = 'svn export ';
+
+        $svn = $builder->getConfig('svn');
+        if (!is_null($svn)) {
+	    foreach ($svn as $key => $value)
+		$cmd .= ' --' . $key . ' ' . $value . ' ';
+        }
+
+        $depth = $builder->getConfig('clone_depth');
+
+        if (!is_null($depth)) {
+            $cmd .= ' -q --depth ' . intval($depth) . ' ';
+        }
+
+        $cmd .= ' -b %s %s "%s"';
+
+        if (!IS_WIN) {
+            $cmd = 'export SVN_SSH="'.$sshWrapper.'" && ' . $cmd;
```

Comment:
If you keep the concatenation, there should be spaces around the `.` here, or use string interpolation